crates/admin/src/cluster_controller/scheduler.rs

```diff
@@ -176,6 +177,10 @@ impl<T: TransportConnect> Scheduler<T> {
                 }
             }
 
+            if !is_active_controller {
+                break;
+            }
```

Comment:
I think this works and I don't want to ask to do it differently now. What I am wondering, though, is whether making the leader state more explicit would help with cleaner code. If, for example, we would only create the `Scheduler` and `LogsController` when we are the leader this wouldn't be necessary. This would then also establish a clear lifecycle (e.g. when becoming the leader then we would take the current `PartitionTable`, `LogsConfiguration` and initialize the controllers) and makes clear which are the actions that a leader needs to do and what every follower. The downside would be if the init step when becoming the leader is expensive and there are many quick leader changes. In this case, the current way would probably behave more graceful.

Reply:
I think I need to withdraw the correctness point. W/o the lower part, we wouldn't update the internal hold `scheduling_plan`. So if we now become leader, then we would start with the `SchedulingPlan` that is based on before this logs update. Differently said, with this if condition it would be equivalent to not call this method if we are not the leader.

Reply:
My thinking was that if we become leader this will trigger a change in cluster state which will rebuild the scheduling plan based on latest known state.

I will definitely revise the code again taking that into account.

Reply:
Maybe tracking the leader state in a state machine would be cleaner for reasoning and readability even if is a little bit slower to switch to leader mode?